partiql-parser/src/scanner.rs

```diff
@@ -123,6 +137,18 @@ fn normalize_quoted_ident(raw_text: &str) -> Cow<str> {
         .into()
 }
 
+fn parse_num<T, R, E>(pair: Pair<R>) -> ParserResult<T>
+where
+    T: Num<FromStrRadixErr = E>,
+    R: RuleType,
+    E: std::fmt::Display,
+{
+    match T::from_str_radix(pair.as_str(), 10) {
+        Ok(value) => Ok(value),
+        Err(e) => pair.syntax_error(format!("Could not parse number {}: {}", pair.as_str(), e)),
+    }
+}
```

Comment:
This just makes it easy to parse and translate errors to syntax error from the `Pair` that represents the token for the number, whether it is a decimal or integer.